ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonParser.java

```diff
@@ -577,7 +582,8 @@ public Duration readDuration() {
    * @return The Period value parsed.
    */
   public Period readPeriod() {
-    if(DEBUG) logger.log(Level.FINEST,"readPeriod " + currentEvent);
+    if (logger.isLoggable(Level.FINEST))
```

Comment:
Is this check needed?

Reply:
Yes. We do not know what  ""readPeriod " + currentEvent" can lead to. When we call log(), Java will call toString() on currentEvent and then concatenate. That operations can be expensive. 
To be sure, that this will not trigger any  computation. Unless we know that the result will be  actually logged , we surround the 
call by this test.